ariwete/pkg/config/config.go

```diff
@@ -207,7 +211,55 @@ func (c *Config) FindSetupFiles(paths []string) (*map[string]CISetup, error) {
 				return nil, err
 			}
 		}
+		err := c.ValidateCISetup(setupFile, setup)
+		if err != nil {
+			return nil, err
+		}
 		setups[path] = setup
 	}
 	return &setups, nil
 }
+
+func (c *Config) ValidateCISetup(path string, setup CISetup) error {
+	var sb strings.Builder
+	for field := range setup {
+		value, exists := c.CISetupDefaults[field]
+		if strings.HasPrefix(field, "_") {
+			// This is a comment field, no need to validate.
+			continue
+		}
+
+		// Check if the field exists.
+		if !exists {
+			sb.WriteString(fmt.Sprintf("%v: '%v' does not exist.\n", path, field))
+			sb.WriteString("\n")
+			sb.WriteString("These are the valid fields:\n")
```

Comment:
design question: Would it make sense to report all extra fields at once and move the list of valid fields outside the loop? This might reduce the number of debugging iterations.

Reply:
Yeah, that makes sense. I did the "fail on first error" approach since it was easier to implement. But I guess the function could return a list of errors instead of a single error.